src/ucp/wireup/wireup.c

```diff
@@ -702,27 +696,25 @@ ucp_wireup_process_ep_check(ucp_worker_h worker, ucp_ep_h ep,
     status = ucp_wireup_init_lanes_by_request(worker, reply_ep, 0,
                                               remote_address, addr_indices);
     if (status != UCS_OK) {
-        goto failed_destroy_ep;
+        goto destroy_ep;
     }
 
     ucp_ep_update_remote_id(reply_ep, msg->src_ep_id);
     status = ucp_wireup_msg_send(reply_ep, UCP_WIREUP_MSG_EP_REMOVED,
                                  &ucp_tl_bitmap_min, NULL);
     if (status != UCS_OK) {
-        goto failed_destroy_ep;
+        goto destroy_ep;
     }
 
     req = ucp_ep_flush_internal(reply_ep, UCP_REQUEST_FLAG_RELEASED,
                                 &ucp_request_null_param, NULL,
                                 ucp_ep_register_disconnect_progress, "close");
-    if (!UCS_PTR_IS_PTR(req)) {
-        ucp_ep_disconnected(ep, 0);
+    if (UCS_PTR_IS_PTR(req)) {
+        return;
     }
 
-    return;
-
-failed_destroy_ep:
-    ucp_ep_destroy_internal(reply_ep);
+destroy_ep:
```

Comment:
why need to rename/refactor this func?

Reply:
reduce exit points from `3` to `2`, because we can use only `ucp_ep_disconnected()` instead of `ucp_ep_destroy_internal()`/`ucp_ep_disconnected()`
+ it fixes issue when destroying EP=`NULL`